xds/internal/xdsclient/client.go

```diff
@@ -42,17 +43,37 @@ type clientImpl struct {
 	done   *grpcsync.Event
 	config *bootstrap.Config
 
-	controller controllerInterface
-
-	logger *grpclog.PrefixLogger
-	pubsub *pubsub.Pubsub
+	// authorityMu protects the authority fields. It's necessary because an
+	// authority is created when it's used.
+	authorityMu sync.Mutex
+	// authorities is a map from ServerConfig to authority. So that
+	// different authorities sharing the same ServerConfig can share the
+	// authority. The key is ServerConfig.String().
+	//
+	// An authority is either in authorities, or idleAuthorities,
+	// never both.
```

Comment:
A47 says the following:

The top-level map in the cache is keyed by authority. There should be a separate entry for old-style resource names (either using some special map key, being careful to avoid collisions, or stored outside of the map). New-style resource names without any authority will use the empty string as the key

How are we doing this?

Reply:
In this implementation, the key of the map is the config `ServerConfig.String()`, not the authority name.

old-style resources use the top level config, new style resource names find the config from the bootstrap map.